test/PowerShellEditorServices.Test/Refactoring/RefactorFunctionTests.cs

```diff
@@ -0,0 +1,250 @@
+using Xunit;
+using Microsoft.PowerShell.EditorServices.Services.Symbols;
+using Microsoft.PowerShell.EditorServices.Refactoring;
+using PowerShellEditorServices.Test.Shared.Refactoring.Functions;
+
+namespace PowerShellEditorServices.Test.Refactoring
+{
+    [Trait("Category", "RefactorFunction")]
+    public class RefactorFunctionTests : IAsyncLifetime
+
+    {
+        private PsesInternalHost psesHost;
+        private WorkspaceService workspace;
+        public async Task InitializeAsync()
+        {
+            psesHost = await PsesHostFactory.Create(NullLoggerFactory.Instance);
+            workspace = new WorkspaceService(NullLoggerFactory.Instance);
+        }
+
+        public async Task DisposeAsync() => await Task.Run(psesHost.StopAsync);
+        private ScriptFile GetTestScript(string fileName) => workspace.GetFile(TestUtilities.GetSharedPath(Path.Combine("Refactoring", "Functions", fileName)));
+
+        internal static string GetModifiedScript(string OriginalScript, ModifiedFileResponse Modification)
+        {
+
+            string[] Lines = OriginalScript.Split(
+                            new string[] { Environment.NewLine },
+                            StringSplitOptions.None);
+
+            foreach (TextChange change in Modification.Changes)
+            {
+                string TargetLine = Lines[change.StartLine];
+                string begin = TargetLine.Substring(0, change.StartColumn);
+                string end = TargetLine.Substring(change.EndColumn);
+                Lines[change.StartLine] = begin + change.NewText + end;
+            }
+
+            return string.Join(Environment.NewLine, Lines);
+        }
+
+        internal static string TestRenaming(ScriptFile scriptFile, RenameSymbolParams request, SymbolReference symbol)
+        {
+
+            //FunctionRename visitor = new(symbol.NameRegion.Text,
+            //                            request.RenameTo,
+            //                            symbol.ScriptRegion.StartLineNumber,
+            //                            symbol.ScriptRegion.StartColumnNumber,
+            //                            scriptFile.ScriptAst);
+            //                            scriptFile.ScriptAst.Visit(visitor);
+            IterativeFunctionRename iterative = new(symbol.NameRegion.Text,
+                                        request.RenameTo,
+                                        symbol.ScriptRegion.StartLineNumber,
+                                        symbol.ScriptRegion.StartColumnNumber,
+                                        scriptFile.ScriptAst);
+            iterative.Visit(scriptFile.ScriptAst);
+            //scriptFile.ScriptAst.Visit(visitor);
+            ModifiedFileResponse changes = new(request.FileName)
+            {
+                Changes = iterative.Modifications
+            };
+            return GetModifiedScript(scriptFile.Contents, changes);
+        }
+
+        [Fact]
+        public void RefactorFunctionSingle()
```

Comment:
Style nit: This can come much later, but it may make sense to reduce deduplication for tests that don't have special requirements by making this a data driven test using TheoryData or MemberData
https://www.milanjovanovic.tech/blog/creating-data-driven-tests-with-xunit#writing-parameterized-tests-with-theorydata

Reply:
I've ended up reworking all the tests to be parameterized now after reading the article you've linked it makes sense and considering how I'm expecting more edge cases, makes it easier to make new ones